src/main/java/database/factories/EventDataFactory.java

```diff
@@ -10,11 +10,28 @@ public class EventDataFactory {
      * @return EventData with all instance attributes converted from jsonObject
      */
     public EventData createEventData(JSONObject jsonObject) {
-        return new EventData(
-                (String) jsonObject.get("name"),
-                (String) jsonObject.get("type"),
-                (long) jsonObject.get("priority")
-        );
+        String type = (String) jsonObject.get("type");
+        if (type.equals("Encounter")) {
+            return new EventData(
```

Comment:
should use different classes for encounterEventData, itemPickupEventData, etc. 
the one eventData class leaves very little room for expansion, since every future type of event added means more useless parameters in eventData.